src/services/incident.services.js

```diff
@@ -30,6 +30,14 @@ const incidentServices = {
         incident.id = newIncidentId;
         incidents.push(incident);
         return incident;
+    },
+
+    // @describe update a red-flag 
+    // @public api/v1/red-flag/:id
+    updateRedFlagLocation (id, location) {
+        const redFlag = incidents.find(incident => incident.id === +id);
+        redFlag.location = location
+        return redFlag || {};
```

Comment:
Why do you have `||` here?

Reply:
I am using the || in case nothing is found, then it should return an empty object.

Reply:
Error should be thrown when someone attempts to update a record that doesn't exist.

The API call should return a 404 response

Reply:
> Error should be thrown when someone attempts to update a record that doesn't exist.
> 
> The API call should return a 404 response

Yes, I have added it in the controller if a particular is not found.

Reply:
services folder just work business logic, but the controller deal with result produce by the services logic.

Reply:
I get that, but the check if item exists is not in the controller

Reply:
> I get that, but the check if item exists is not in the controller

Is on the controller 

if (!id) {
            return next(new customError(404, 'Id is not found!'));
       }